src/app/client/src/app/modules/shared/components/share-link/share-link.component.ts

```diff
@@ -95,6 +95,15 @@ export class ShareLinkComponent implements OnInit {
     document.execCommand('copy');
   }
 
+  setCopyContentSharingLinkInteractEdata() {
+    const searchInteractEdata = {
+      id: 'content-share-link-btn',
+      type: 'share',
+      pageid: 'resource'
```

Comment:
shoudnt be hardcoded, this is common component and can be used in course page also. 
Please take input like telemetryShareData in ShareLinkComponent. Add object like {"id":"do_2128459444926791681270","type":"Course","ver":"1"}